common/src/definitions/guns.ts

```diff
@@ -1265,3 +1318,31 @@ export const Guns: GunDefinition[] = [
         }
     }
 ];
+
+export const Guns: GunDefinition[] = GunsRaw.map(e => {
+    if (e.dual === undefined) {
+        return [e];
+    }
+
+    const dualDef = mergeDeep({}, e, e.dual) as GunDefinition & { readonly dual?: object, readonly isDual: true };
+    // @ts-expect-error init code
+    delete dualDef.dual;
+    // @ts-expect-error init code
+    delete dualDef.fists;
+    // @ts-expect-error init code
+    delete dualDef.image;
+    // @ts-expect-error init code
+    delete dualDef.casingParticles;
+    // @ts-expect-error init code
+    dualDef.isDual = true;
+    // @ts-expect-error init code
+    dualDef.singleVariant = e.idString;
+    // @ts-expect-error init code
+    dualDef.idString = `dual_${dualDef.idString}`;
+    // @ts-expect-error init code
+    e.dualVariant = dualDef.idString;
+    // @ts-expect-error init code
+    dualDef.name = `Dual ${dualDef.name}`;
```

Comment:
can't you remove some of those ts-expect-error by putting the properties inside mergedeep?